### PR TITLE
chore: refactor exposed function for BiDi

### PR DESCRIPTION
**Pull request by jrandolf-google**
Using one channel instead of three.

Bug: https://github.com/puppeteer/puppeteer/issues/11981

### CODE REVIEW

File: packages/puppeteer-core/src/bidi/ExposedFunction.ts [hunk 1]
```diff
@@ -106,190 +85,144 @@ export class ExposeableFunction<Args extends unknown[], Ret> {
...
     void this[Symbol.asyncDispose]().catch(debugError);
   }
 
   async [Symbol.asyncDispose](): Promise<void> {
-    if (this.#preloadScriptId) {
-      await this.#connection.send('script.removePreloadScript', {
-        script: this.#preloadScriptId,
-      });
+    for (const [frame, script] of this.#scripts) {
+      await frame.browsingContext.removePreloadScript(script);
```
Comment (OrKoN) [inline packages/puppeteer-core/src/bidi/ExposedFunction.ts]: let's do promise all to dispose it quicker

File: packages/puppeteer-core/src/bidi/ExposedFunction.ts [hunk 2]
```diff
@@ -38,65 +32,50 @@ export class ExposeableFunction<Args extends unknown[], Ret> {
...
-    const channelArguments = this.#channelArguments;
+    const channel = {
+      type: 'channel' as const,
+      value: {
+        channel: this.#channel,
+        ownership: Bidi.Script.ResultOwnership.Root,
+      },
+    };
 
     // TODO(jrandolf): Implement cleanup with removePreloadScript.
```
Comment (OrKoN) [inline packages/puppeteer-core/src/bidi/ExposedFunction.ts]: implemented now?

Comment (jrandolf) [reply packages/puppeteer-core/src/bidi/ExposedFunction.ts]: This is for `removeExposeFunction`

**Review by OrKoN — APPROVED** [review]

File: packages/puppeteer-core/src/bidi/ExposedFunction.ts (hunk 2 above)
Comment (jrandolf) [reply packages/puppeteer-core/src/bidi/ExposedFunction.ts]: I've removed the comment. Linking the bug here instead.

**Comment by OrKoN** [comment]
Looks like some test fail in Firefox

**Comment by jrandolf** [comment]
> Looks like some test fail in Firefox

Looks like that test wasn't testing what it needed. We don't guarantee that exposed function calls in a preload script will resolve before navigation finishes. I've fixed the test to actually reflect what its title states.

**Review by OrKoN — APPROVED** [review]

**Pull request merged**